cpp/src/arrow/compute/api_scalar.cc

```diff
@@ -357,6 +357,9 @@ SCALAR_ARITHMETIC_BINARY(Power, "power", "power_checked")
 SCALAR_ARITHMETIC_BINARY(ShiftLeft, "shift_left", "shift_left_checked")
 SCALAR_ARITHMETIC_BINARY(ShiftRight, "shift_right", "shift_right_checked")
 SCALAR_EAGER_BINARY(Atan2, "atan2")
+SCALAR_EAGER_UNARY(Floor, "floor")
+SCALAR_EAGER_UNARY(Ceiling, "ceiling")
```

Comment:
```suggestion
SCALAR_EAGER_UNARY(Ceil, "ceil")
```

Is there a reason to have it "ceiling" instead of "ceil" (the C++ function is ceil, as well as numpy. SQL seems to have both)

Reply:
I agree that `ceil` should be the name. Thanks for this review!

Reply:
@jorisvandenbossche Compute function names (for both C++ API and `CallFunction` name) are inconsistent w.r.t. to short vs. long form. For example, *Ceiling, Negate, Power*, etc. I think this should be revisited in another JIRA issue and will require updating Python and R bindings.

Reply:
I used the short form for compute function names in this PR, and opened [this JIRA](https://issues.apache.org/jira/projects/ARROW/issues/ARROW-13359) to revisit the names of compute functions.